handsontable/src/plugins/base/base.js

```diff
@@ -192,6 +207,41 @@ export class BasePlugin {
     }
   }
 
+  /**
+   * Check if any of the keys defined in `SETTING_KEYS` configuration of the plugin is present in the provided
+   * config object, or if the `SETTING_KEYS` configuration states that the plugin is relevant to the config object
+   * regardless of its contents.
+   *
+   * @private
+   * @param {Handsontable.DefaultSettings} settings The config object passed to `updateSettings`.
+   * @returns {boolean}
+   */
+  #isRelevantToSettings(settings) {
+    if (!settings) {
+      return false;
+    }
+
+    const settingKeys = this.constructor.SETTING_KEYS;
+
+    // If SETTING_KEYS is declared as `true` -> update the plugin regardless of the settings declared in
+    // `updateSettings`.
+    if (settingKeys === true) {
+      return true;
+
+    // If SETTING_KEYS is declared as `false` -> DON'T update the plugin regardless of the settings declared in
+    // `updateSettings`.
+    } else if (settingKeys === false) {
+      return false;
+
+    } else {
+      for (let i = 0; i < settingKeys.length; i++) {
+        if (settings[settingKeys[i]] !== void 0) {
+          return true;
+        }
+      }
+    }
```

Comment:
```suggestion
    // If SETTING_KEYS is declared as `true` -> update the plugin regardless of the settings declared in
    // `updateSettings`.
    // If SETTING_KEYS is declared as `false` -> DON'T update the plugin regardless of the settings declared in
    // `updateSettings`.
    if (typeof settingKeys === 'boolean') {
      return settingKeys;
    }

    for (let i = 0; i < settingKeys.length; i++) {
      if (settings[settingKeys[i]] !== void 0) {
        return true;
      }
    }

    return false;
```
When the `settingKeys` variable is an array and the `if`s in the `for` logic never be true the function returns `undefined`. WDYT about reshaping that lines to remove redundant `if`/`else`s?